documentation/modules/ref-kafka-jmx-options.adoc

```diff
@@ -0,0 +1,61 @@
+
+// Module included in the following assemblies:
+//
+// assembly-deployment-configuration-kafka.adoc
+
+[id='ref-kafka-jmx-options-{context}']
+= JMX configuration
+
+You can configure JMX options by using the `jmxOptions` property in the following resources:
+
+* `Kafka.spec.kafka`
+
+You can configure username and password protection for the JMX port that is opened on the Kafka brokers.
+
+.Securing the JMX Port
+Currently the JMX port can only be secured via username and password.
+To enable security for the JMX port, set the `type` parameter in the `authentication` field to `password`, for example:
+[source,yaml,subs=attributes+]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    jmxOptions:
+      authentication:
+        type: "password"
+    # ...
+  zookeeper:
+    # ...
+----
+In the above example, the JMX port on the Kafka brokers are opened and then the JMX port on the Kafka headless service
+is opened.
+This will allow you to deploy a pod internally into a cluster and obtain JMX metrics by using the headless
+service and specifying which broker you want to address.
+For example, to get metrics from broker 0 we address the headless service by doing the following:
+"_`<cluster-name>`_-kafka-0- _`<cluster-name>`_-_`<headless-service-name>`_" on the JMX port.
+If the JMX port is secured then you can get the username and password by referencing them from the JMX secret in the
+deployment of your pod.
+
+.Using an open JMX port
+To disable security for the JMX port, do not fill in the `authentication` field, for example:[source,yaml,subs=attributes+]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    jmxOptions: {}
+    # ...
+  zookeeper:
+    # ...
+----
+
+This will just open the JMX Port on the headless service and you can follow a similar approach as described above
```

Comment:
Do we want to recommend an approach? Os say under what circumstances you might want to use one or the other?

Reply:
Okay